src/meltano/core/plugin/singer/tap.py

```diff
@@ -431,6 +431,12 @@ async def discover_catalog(
                 Draft4Validator.check_schema(catalog)
         except Exception as err:
             catalog_path.unlink()
+            if isinstance(err, json.JSONDecodeError):
+                logger.error(
+                    "Invalid JSON: %s [...]",
+                    err.doc[err.pos : err.pos + 10],
```

Comment:
Would it be more useful to displace a range centered on `err.pos`?

```suggestion
                    err.doc[max(err.pos - 9, 0) : err.pos + 10]
```

Reply:
Yeah, that makes sense!